pandas/tests/indexes/datetimes/test_indexing.py

```diff
@@ -46,6 +46,14 @@ def test_where_tz(self):
         expected = i2
         tm.assert_index_equal(result, expected)
 
+    @pytest.mark.parametrize('tz', [None, 'UTC', 'US/Eastern'])
+    def test_insert_nat(self, tz):
```

Comment:
You could maybe parametrize the over different NA values here, since this should work for inserting `np.nan` and `None` as well.  For example, on a tz-naive DTI:
```python
In [2]: pd.date_range('20170101', periods=2).insert(0, np.nan)
Out[2]: DatetimeIndex(['NaT', '2017-01-01', '2017-01-02'], dtype='datetime64[ns]', freq=None)

In [3]: pd.date_range('20170101', periods=2).insert(0, None)
Out[3]: DatetimeIndex(['NaT', '2017-01-01', '2017-01-02'], dtype='datetime64[ns]', freq=None)
```

Reply:
If you do that, I suppose you could delete these lines in `test_insert` below, as to not duplicate things:

https://github.com/pandas-dev/pandas/blob/ac8ac158377d60c8c8e10e6905d49d48bbe80855/pandas/tests/indexes/datetimes/test_indexing.py#L148-L153